src/components/Forms/SignIn.tsx

```diff
@@ -0,0 +1,183 @@
+import { useAppDispatch } from '../../hooks/useAppDispatch';
+import { LocaleContext } from '../LocaleContext/LocaleContext';
+import SubSide from './SubSide';
+import { object, string } from 'yup';
+import { SignInForm } from '../../dto/types';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { signInWithEmailAndPassword } from 'firebase/auth';
+import { login } from '../../store/slices/userSlice';
+import { Navigate } from 'react-router-dom';
+import { Paths } from '../../dto/constants';
+import { FirebaseError } from 'firebase/app';
+import { auth } from '../../services/firebase';
+import { toast } from 'react-toastify';
+import Loader from '../Loader/Loader';
+
+const SignIn: FC = () => {
+  const isLoginPath = useAppSelector(authPathSelector);
+  const dispatch = useAppDispatch();
+  const { spellingList } = useContext(LocaleContext);
+  let errorCode: string;
+  const schemaLogin = object({
+    email: string()
+      .required(`${spellingList.forms.requiredError}`)
+      .email(`${spellingList.forms.emailError}`),
+    password: string()
+      .required(`${spellingList.forms.requiredError}`)
+      .matches(/^(?=.*[0-9])/, `${spellingList.forms.passwordErrorDigit}`)
+      .matches(
+        /^(?=.*[A-Za-zА-Яа-я])/,
+        `${spellingList.forms.passwordErrorLetter}`
+      )
+      .matches(
+        /^(?=.*[!@#%&$^*()?><|+=])/,
+        `${spellingList.forms.passwordErrorChar}`
+      )
+      .matches(
+        /^[\w\Wa-zA-Z0-9!@#%&$^*()?><|+=]{8,}$/,
+        `${spellingList.forms.passwordErrorCount}`
+      ),
+  });
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<SignInForm>({
+    mode: 'onSubmit',
+    resolver: yupResolver(schemaLogin),
+    reValidateMode: 'onSubmit',
+  });
+
+  const [loading, setLoading] = useState(false);
+
+  const onSubmit = async (data: SignInForm) => {
+    setLoading(true);
+    signInWithEmailAndPassword(auth, data.email, data.password)
+      .then(() => {
+        setLoading(false);
+        dispatch(login());
+        toast.success(`${spellingList.forms.success}`, { draggable: true });
+        <Navigate to={Paths.MAIN} replace />;
+      })
+      .catch((error) => {
+        setLoading(false);
+        if (error instanceof FirebaseError) {
+          switch (error.code) {
+            case 'auth/internal-error':
+              errorCode = `${spellingList.forms.firebaseErrorInternal}`;
+              break;
+            case 'auth/invalid-credential':
+              errorCode = `${spellingList.forms.firebaseErrorInvalidCredential}`;
+              break;
+            case 'auth/network-request-failed':
+              errorCode = `${spellingList.forms.firebaseErrorNetwork}`;
+              break;
+            case 'auth/too-many-requests':
+              errorCode = `${spellingList.forms.firebaseErrorMany}`;
+              break;
+            default:
+              errorCode = error.code;
```

Comment:
Noo need to use string template literal on every string. you can use it as it is